### ISSUE TITLE
Class files are not compatible with Java 7

### ISSUE THREAD

**Issue by wytten** [dialex/JCDP#8]
If the maven-compiler-plugin were changed to source 1.7 target 1.7 the JCDP jar wouldn't require Java 8

**Comment by dialex** [comment]
Sorry for the late reply, I did not get a notification -- it won't happen again 😉

I'll just double-check if any of the dependencies requires Java 8, if not I'll downgrade to Java 7.

**Comment by dialex** [comment]
Yup **Jansi requires Java 6** and **jUnit requires Java 5**, so we can go as low as 6.

**Comment by wytten** [comment]
Thank you kind sir!

**Comment by dialex** [comment]
New tag published on [GitHub](https://github.com/dialex/JCDP/releases/tag/v2.0.2) and still pending on [Maven](https://mvnrepository.com/artifact/com.diogonunes/JCDP/2.0.2)

**Comment by wytten** [comment]
There's a problem with the pom, it is still hardcoded for 1.8 and ignoring the maven.compiler.* properties that have been added:
`linux% unzip ./repo/com/diogonunes/JCDP/2.0.2/JCDP-2.0.2.jar
...
linux% file com/diogonunes/jcdp/color/ColoredPrinter.class
com/diogonunes/jcdp/color/ColoredPrinter.class: compiled Java class data, version 52.0 (Java 1.8)`

**Comment by dialex** [comment]
Oh noes! Thanks for the heads up, I'll have a look at it asap.

**Comment by dialex** [comment]
Found it! The issue was this block

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-compiler-plugin</artifactId>
    <version>3.5.1</version>
    <configuration>
        <source>1.8</source> <!-- should be 6 -->
        <target>1.8</target> <!-- should be 6 -->
    </configuration>
</plugin>
```

Now I have to find a way to submit a (hot)fix or patch to Nexus. I would like to avoid creating/pushing another release. Makes sense?

**Comment by wytten** [comment]
Yes I think the fix should be

-                    <source>1.8</source>
-                    <target>1.8</target>
+                    <source>${maven.compiler.source}</source>
+                    <target>${maven.compiler.target}</target>

Thanks

**Comment by dialex** [comment]
I've released [v2.0.3.1](https://github.com/dialex/JCDP/releases/tag/v2.0.3.1) containing your fix. It may take a while to appear on Maven.

**Comment by wytten** [comment]
Great, thanks!
